File: backend/src/main/java/com/springbootjwtpostgres/backend/product/ProductSearchCriteria.java
package com.springbootjwtpostgres.backend.product;

import lombok.Data;

import java.util.Date;

@Data
public class ProductSearchCriteria {
    private String productName;
    private String productCode;
    private double productPriceInFrom;
    private double productPriceInTo;
    private double productPriceOutFrom;
    private double productPriceOutTo;
    private String note;
    private Date createdAtFrom;
    private Date createdAtTo;
}

File: frontend/src/app/models/product/ProductSearchCriteria.ts
export class ProductSearchCriteria {
  productName: string;
  productCode: string;
  productPriceInFrom: number;
  productPriceInTo: number;
  productPriceOutFrom: number;
  productPriceOutTo: number;
  note: string;
  createdAtFrom: Date;
  createdAtTo: Date;

  constructor(
    productName: string,
    productCode: string,
    productPriceInFrom: number,
    productPriceInTo: number,
    productPriceOutFrom: number,
    productPriceOutTo: number,
    note: string,
    createdAtFrom: Date,
    createdAtTo: Date
  ) {
    this.productName = productName;
    this.productCode = productCode;
    this.productPriceInFrom = productPriceInFrom;
    this.productPriceInTo = productPriceInTo;
    this.productPriceOutFrom = productPriceOutFrom;
    this.productPriceOutTo = productPriceOutTo;
    this.note = note;
    this.createdAtFrom = createdAtFrom;
    this.createdAtTo = createdAtTo;
  }
}
